src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -300,6 +304,7 @@ public CSharpOperationFactory(SemanticModel semanticModel)
                 case BoundKind.StackAllocArrayCreation:
                 case BoundKind.TypeExpression:
                 case BoundKind.TypeOrValueExpression:
+                case BoundKind.UnconvertedCollectionLiteralExpression: // PROTOTYPE: Is this correct?
```

Comment:
No, I would expect this to be either a ICollectionLiteralOperation with a null type, or a `IInvalidOperation`. `NoneOperation`s are just for things that we do not support, and we should not be adding anything to this list.